esphomeyaml/components/sensor/total_daily_energy.rst

```diff
@@ -17,9 +17,9 @@ daily energy usage in ``Wh`` or ``kWh``.
     # Example configuration entry
     sensor:
       - platform: total_daily_energy
-        pin: 12
         name: "Total Daily Energy"
         power_id: my_power
+        time_id: my_time
```

Comment:
This option is optional. It is auto-detected if you have a time component set up.